docs/markdown/Reference-manual.md

```diff
@@ -2724,6 +2724,24 @@ The following methods are defined for all [`feature` options](Build-options.md#f
 - `enabled()`: returns whether the feature was set to `'enabled'`
 - `disabled()`: returns whether the feature was set to `'disabled'`
 - `auto()`: returns whether the feature was set to `'auto'`
+- `allowed()` *(since 0.58.0)*: returns whether the feature was set to `'enabled'` or `'auto'`
+- `may_disable_if(value)` *(since 0.58.0)*: returns the feature, with
+  `'auto'` converted to `'disabled'` if value is true.
+- `require(value, error_message: '')` *(since 0.58.0)*: returns
+  the object itself if the value is true; an error if the object is
+  `'enabled'` and the value is false; a disabled feature if the object
+  is `'auto'` or `'disabled'` and the value is false.
```

Comment:
So this is the same as `may_disable_if()` except that it is an error if the user set that feature to enabled when it cannot be supported, right?

This makes me wonder if we even need `may_disable_if()`, the way I see feature options is that if user enable one it's should be a guarantee that if for any reason that cannot be satisfied they get an error and not a silently disabled feature.

Reply:
`may_disable_if()` does not degrade `enabled` to `disabled`, only `auto`. This is why it has a slightly more complicated name than `disabled_if()`.

The design should cover exactly what you mention here. The idea is that you use

```
    libone = dependency('one', requires: get_option('one'))
    libtwo = dependency('two',
       requires: get_option('two').may_disable_if(not libone.found())
```

This way:
* with `-Dtwo=auto` libtwo won't even be searched unless libone was found
* with `-Dtwo=enabled` it will be an error if libtwo is absent, even if the build will end up not using it because libone is absent.

Reply:
Hmm, I'm not really convinced by `may_disable_if()` tbh, do you have real use-case in qemu for that? It feels like `.require()` is easy and intuitive, but `may_disable_if()` is for subtle corner case that I'm not even sure to really grasp.

Reply:
What about renaming `value` to `condition` here for clarity?

Could also change the error_message kwarg to be simply an optional 2nd positional argument, just like `assert(condition, 'optional message')`?

Maybe add working like "This describes a condition required to allow enabling this feature" ?